src/compile/compile.ts

```diff
@@ -85,14 +92,6 @@ function assemble(model: Model, topLevelProperties: TopLevelProperties) {
         model.assembleSelectionTopLevelSignals([])
       )
     ),
-
-    // FIXME: get rid of the top-level `nested-main-group`
-    // HACK: this is a hack to temporarily make selections works as
-    // 1) Currently, some selection's signals rely on the main group's scope to shadow duplicate names.
-    // 2) Selection predicate depends on parent reference which may not exist for top-level mark.
-    ...assembleNestedMainGroup(model),
```

Comment:
D'oh, think I temporarily undid it's removal during some churn and forgot to redo it. Fixed.